### PR TITLE
Serialization refactoring

### PR DESCRIPTION
**Pull request by vkagamlyk**
Performance enhancements for Golang Driver

### CODE REVIEW

**Comment by lyndonbauto** [comment]
For TinkerPop we usually try to keep the commit log pretty clean, so I will squash the commit log on this pull request to do that before merging.

File: gremlin-go/driver/serializer.go [hunk 1]
```diff
@@ -165,135 +173,147 @@ func uuidToBigInt(requestID uuid.UUID) big.Int {
...
+			bindingType:           bindingWriter,
+			mapType:               mapWriter,
+			listType:              listWriter,
+		}
+	}
+}
 
-	return msg, nil
+func initDeserializers() {
+	// todo: lock or don't care?
```
Comment (lyndonbauto) [inline gremlin-go/driver/serializer.go:281]: Good question. 

I think worst case we have a situation of overwriting the deserializers with an equivalent object, so effectively no change.

I don't see a need to lock.

**Review by lyndonbauto — APPROVED** [review]

**Comment by lyndonbauto** [comment]
I have looked at this at a previous date and am good with the changes.

VOTE+1

File: gremlin-go/driver/serializer.go (hunk 1 above)
Comment (L0Lmaker) [reply gremlin-go/driver/serializer.go:281]: remove comment

**Comment by spmallette** [comment]
VOTE +1

**Comment by mikepersonick** [comment]
LGTM VOTE+1

**Comment by lyndonbauto** [comment]
Thanks for the change and thanks for the vote. I will propagate this.

**Pull request merged**